Dip/Dip/Dip.swift

```diff
@@ -42,8 +42,7 @@ public class DependencyContainer {
     case Int(IntegerLiteralType)
   }
   
-  private var dependencies = [DefinitionKey : Definition]()
-  private var lock: OSSpinLock = OS_SPINLOCK_INIT
```

Comment:
Yeah but removing it breaks thread-safety. Better replace it with a `NSRecursiveLock` instead maybe?